core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala

```diff
@@ -114,7 +114,7 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
 
     context.addTaskCompletionListener[Unit] { _ =>
       writerThread.shutdownOnTaskCompletion()
-      if (!reuseWorker || !released.get) {
+      if (!reuseWorker || released.compareAndSet(false, true)) {
```

Comment:
I think `released` variable here indices if `releasePythonWorker` is actually called or not. Do you mind if I ask to elaborate why this `released` it set to `true` here as well?